lib/trento/software_updates/discovery.ex

```diff
@@ -80,32 +80,54 @@ defmodule Trento.SoftwareUpdates.Discovery do
   end
 
   def discover_host_software_updates(host_id, fully_qualified_domain_name) do
-    with {:ok, system_id} <- get_system_id(fully_qualified_domain_name),
-         {:ok, relevant_patches} <- get_relevant_patches(system_id),
-         :ok <-
-           host_id
-           |> build_discovery_completion_command(relevant_patches)
-           |> commanded().dispatch() do
-      {:ok, host_id, system_id, relevant_patches}
-    else
-      error ->
+    case discover_relevant_patches(fully_qualified_domain_name) do
+      {:ok, system_id, relevant_patches} ->
+        determine_health_and_dispatch_completion_command(host_id, system_id, relevant_patches)
+
+      {:error, discovery_error} = error ->
         Logger.error(
           "An error occurred during software updates discovery for host #{host_id}:  #{inspect(error)}"
         )
 
-        {:error, error}
+        dispatch_completion_command_on_discovery_error(host_id, discovery_error)
```

Comment:
I don't understand a thing :sweat_smile: 
Wasn't as simple as putting the the dispatch with the `unknown` state in the `else` side?

```
commanded().dispatch(
      CompleteSoftwareUpdatesDiscovery.new!(%{
        host_id: host_id,
        health: SoftwareUpdatesHealth.unknown()
    })
)

error
```